src/QMCWaveFunctions/lcao/LCAOrbitalBuilder.cpp

```diff
@@ -698,7 +705,25 @@ bool LCAOrbitalBuilder::putPBCFromH5(LCAOrbitalSet& spo, xmlNodePtr coeff_ptr)
   spo.setIdentity(false);
   hdf_archive hin(myComm);
 
-  xmlNodePtr curtemp = coeff_ptr->parent->parent->parent;
+  xmlNodePtr curtemp = coeff_ptr;
+
+  std::string xmlTag("determinantset");
+  std::string MSDTag("sposet");
+  std::string SDTag("determinant");
+  std::string curname;
+
+  do
+  {
+    std::stringstream ss;
+    curtemp = curtemp->parent;
+    ss << curtemp->name;
+    ss >> curname;
+    if (curname == MSDTag)
+      MultiDet = true; ///Used to know if running an MSD calculation - needed for order of Orbitals.
+    if (curname == SDTag)
+      MultiDet = false;
+
+  } while (xmlTag != curname);
```

Comment:
I don't really like the potential for an infinite loop here.  Also, this depends on XML input order.

Reply:
If neither "sposet" or "determinant" is written in the XML, this will be an infinite loop. While this might be error trapped earlier (is it?) I think this code should properly error trap the end of the stringstream and the tag not found condition. (abort("Did not find either sposet or determinant in determinantset input.")) We can definitely afford to write conservative/safe code in this initialization part.

Reply:
This is actually a good place to discuss this as there are multiple ways to handle this part, which will also add to @jtkrogel previous remark. 
This code here comes from the need to pick up the order of the orbitals to read. However, @ye-luo mentioned a good way to do it that would fix this point and the previous remark from everyone that we should be also able to read the orbitals in whatever scheme we want either for MSD or SD  and just to be able to read the coordinate of the supertwist!!!! .

One way to do it is to introduce a new tag in the input : coeff_order="unsorted/1excitation/2excitations/etc.." or whatever tags we want. As a second step, spo and determinantset should be merged  in the builder and the choice of building handled by this tag (These are suggestions for another PR of course.. ). 

In the meantime, @prckent  is right, It is impossible to reach this part of the code without having either sposet or determinantset. the previous implementation was actually dangerous as I implemented it with a hard:  curtemp = cur->parent->parent->parent->parent; any modification or change in the xml would have broken this chain (which broke for MSD). 
Because sposet and determinantset are different, the number of jumps back in the xml file were  different. In this case I am sure to always find the return. As a safe implementation, I can add a  condition (if (xmlTag="qmcsystem") = > abbort("Could not find either tags - Check your Wavefunction input") ). Would that be satisfying?

Reply:
@anbenali how about something like:

readit=false;
while (ss) {
 ... 
if (curname==somethingtag) { readit=true; mode=something; break;}
...
}
if (!readit) abort("Did not find expected tag");

Reply:
Assumes the stream becomes null when the end or some error is found